src/components/UserLinks/UserLinks.js

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect, useState } from 'react';
+import { Link } from "react-router-dom";
+import firebase from '../../utils/firebaseConfig';
+
+const initialDisplayedProfileActionsList = [];
+
+const UserLinks = () => {
+
+  const [areProfileActionsDisplayed, setAreProfileActionsDisplayed] = useState(
+    false
+  );
+  const handleCloseProfileLinks = () => {
+    setAreProfileActionsDisplayed(false);
+    setDisplayedProfileActionsList(initialDisplayedProfileActionsList);
+  }
+
+  const [displayedProfileActionsList, setDisplayedProfileActionsList] = useState(initialDisplayedProfileActionsList); 
```

Comment:
Sorry but I don't understand what it does, can you explain ? I tried to remove it completely and it seems to work fine

``` javascript
const UserLinks = () => {

  const [areProfileActionsDisplayed, setAreProfileActionsDisplayed] = useState(
    false
  );
  const handleCloseProfileLinks = () => {
    setAreProfileActionsDisplayed(false);
  }

  return (
      <ul className="profil">
        {!areProfileActionsDisplayed ? (
          <li onClick={() => setAreProfileActionsDisplayed(true)}>
            <i className="far fa-user"></i>Profil
          </li>
        ) :
        <>
          <li>
            <Link to="/profile">
              <i className="fas fa-cog" />Gérer mon profil
            </Link>
          </li>
          <li onClick={() => firebase.auth().signOut()}>
            <i className="fas fa-sign-out-alt" />Se déconnecter
          </li>
          <li onClick={handleCloseProfileLinks}>
            <i className="fas fa-times" />Fermer
          </li>
        </>
        }
    </ul>
  );
};
```

Reply:
> Sorry but I don't understand what it does, can you explain ? I tried to remove it completely and it seems to work fine
> 
> ```js
> const UserLinks = () => {
> 
>   const [areProfileActionsDisplayed, setAreProfileActionsDisplayed] = useState(
>     false
>   );
>   const handleCloseProfileLinks = () => {
>     setAreProfileActionsDisplayed(false);
>   }
> 
>   return (
>       <ul className="profil">
>         {!areProfileActionsDisplayed ? (
>           <li onClick={() => setAreProfileActionsDisplayed(true)}>
>             <i className="far fa-user"></i>Profil
>           </li>
>         ) :
>         <>
>           <li>
>             <Link to="/profile">
>               <i className="fas fa-cog" />Gérer mon profil
>             </Link>
>           </li>
>           <li onClick={() => firebase.auth().signOut()}>
>             <i className="fas fa-sign-out-alt" />Se déconnecter
>           </li>
>           <li onClick={handleCloseProfileLinks}>
>             <i className="fas fa-times" />Fermer
>           </li>
>         </>
>         }
>     </ul>
>   );
> };
> ```

![2021-03-14_19h48_08](https://user-images.githubusercontent.com/4164295/111080355-6706f180-84fe-11eb-82ca-47215acd913f.gif)

When you click on the "Profil" button each element of the list is displayed one by one. Maybe the delay between each <LI> was so short then I updated it to 200ms.